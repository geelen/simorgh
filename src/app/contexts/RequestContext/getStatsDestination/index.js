/* Returns stats destnation for ATI based on origin, service and env
   see table on this issue https://github.com/bbc/simorgh/issues/2995
*/
const getStatsDestination = ({ isUK = true, env = 'test', service }) => {
  let destination = '';
  switch (service) {
    case 'news':
      // checks if "news" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';
      break;
    case 'cymrufyw':
    case 'naidheachdan':
      // checks if "cymrufyw" or "naidheachdan" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_LANGUAGES_PS' : 'NEWS_LANGUAGES_GNL';
      break;
    case 'japanese':
      destination = 'NEWS_LANGUAGES_GNL';
      break;
    case 'scotland':
      destination = 'PLACEHOLDER';
      break;
    case 'archive':
      destination = 'BBC_ARCHIVE_PS';
      break;
    default:
      destination = 'WS_NEWS_LANGUAGES';
  }
  return env === 'live' ? destination : `${destination}_TEST`;
};

export default getStatsDestination;
