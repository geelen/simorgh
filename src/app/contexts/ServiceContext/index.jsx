import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
