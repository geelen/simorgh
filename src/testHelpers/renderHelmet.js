/*
 * Similar to this problem https://github.com/testing-library/react-testing-library/issues/402
 * This helper was created to solve the problem of rendering helmet/head contents in snapshots.
 * Pass in a component that uses helmet somewhere in the component tree.
 * The full html tree is rendered and returned which you can then use to snapshot helmet/head contents.
 * Example of use in a test file:

 it('should render correctly', async () => {
    const html = await renderHelmet(<SomeComponent {...someProps} />);
    expect(html).toMatchSnapshot();
  });

 */

import { render, waitForDomChange } from '@testing-library/react';

export default async component => {
  render(component);

  await waitForDomChange();

  const htmlElement = document.querySelector('html');

  const helmetElements = document.querySelectorAll(
    '[data-react-helmet="true"]',
  );

  const removeHelmetAttributes = el => el.removeAttribute('data-react-helmet'); // remove react-helmet attribute noise from elements

  removeHelmetAttributes(htmlElement); // remove react-helmet attribute noise from elements

  Array.from(helmetElements).forEach(removeHelmetAttributes);

  return document.querySelector('html');
};
