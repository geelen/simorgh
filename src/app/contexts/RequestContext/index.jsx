import React from 'react';
import { bool, node, oneOf, string, number } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';
import getMetaUrls from './getMetaUrls';
import variantPropType from '../../models/propTypes/variants';

export const RequestContext = React.createContext({});

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  statusCode,
  previousPath,
  pathname,
  variant,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    isAmp,
    platform,
    statsDestination,
    statsPageIdentifier,
    statusCode,
    previousPath,
    variant,
    ...getMetaUrls(origin, pathname),
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf([
    'article',
    'frontPage',
    'media',
    'error',
    'MAP',
    'FIX',
    'STY',
    'PGL',
  ]).isRequired,
  service: string.isRequired,
  statusCode: number,
  pathname: string.isRequired,
  previousPath: string,
  variant: variantPropType,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  statusCode: null,
  previousPath: null,
  variant: null,
};
