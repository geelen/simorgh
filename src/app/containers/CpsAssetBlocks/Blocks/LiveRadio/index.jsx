import React, { useContext } from 'react';
import { string } from 'prop-types';
import {
  CanonicalMediaPlayer,
  AmpMediaPlayer,
} from '@bbc/psammead-media-player';
import { RequestContext } from '../../../../contexts/RequestContext';

const LiveRadioContainer = ({ idAttr, externalId, id }) => {
  const { platform } = useContext(RequestContext);

  const MediaPlayer = {
    canonical: CanonicalMediaPlayer,
    amp: AmpMediaPlayer,
  }[platform];

  if (!MediaPlayer || !externalId || !id) return null;

  return (
    <MediaPlayer
      showPlaceholder={false}
      src={`/ws/av-embeds/media/${externalId}/${id}`}
      id={idAttr}
      skin="audio"
    />
  );
};

LiveRadioContainer.propTypes = {
  idAttr: string,
  externalId: string.isRequired,
  id: string.isRequired,
};

LiveRadioContainer.defaultProps = {
  idAttr: null,
};

export default LiveRadioContainer;
