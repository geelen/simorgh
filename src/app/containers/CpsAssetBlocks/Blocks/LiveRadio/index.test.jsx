import React from 'react';
import { shouldMatchSnapshot, isNull } from '@bbc/psammead-test-helpers';
import { RequestContext } from '../../../../contexts/RequestContext';
import { suppressPropWarnings } from '../../../../../testHelpers';
import LiveRadio from '.';

describe('CpsAssetBlocks LiveRadio', () => {
  shouldMatchSnapshot(
    'should render correctly for canonical',
    <RequestContext.Provider value={{ platform: 'canonical' }}>
      <LiveRadio uuid="uuid" idAttr="idAttr" externalId="externalId" id="id" />
    </RequestContext.Provider>,
  );

  // TODO: remove the need for this suppressPropWarnings
  suppressPropWarnings(['placeholderSrc', 'undefined']);

  shouldMatchSnapshot(
    'should render correctly for amp',
    <RequestContext.Provider value={{ platform: 'amp' }}>
      <LiveRadio uuid="uuid" idAttr="idAttr" externalId="externalId" id="id" />
    </RequestContext.Provider>,
  );

  describe('when platform is unknown', () => {
    suppressPropWarnings(['text', 'undefined']);

    isNull(
      'should render null',
      <RequestContext.Provider value={{ platform: 'foobar' }}>
        <LiveRadio
          uuid="uuid"
          idAttr="idAttr"
          externalId="externalId"
          id="id"
        />
      </RequestContext.Provider>,
    );
  });

  describe('when id isnt provided', () => {
    suppressPropWarnings(['id', 'undefined']);

    isNull(
      'should render null',
      <RequestContext.Provider value={{ platform: 'foobar' }}>
        <LiveRadio uuid="uuid" idAttr="idAttr" externalId="externalId" />
      </RequestContext.Provider>,
    );
  });

  describe('when externalId isnt provided', () => {
    suppressPropWarnings(['externalId', 'undefined']);

    isNull(
      'should render null',
      <RequestContext.Provider value={{ platform: 'foobar' }}>
        <LiveRadio uuid="uuid" idAttr="idAttr" id="id" />
      </RequestContext.Provider>,
    );
  });
});
