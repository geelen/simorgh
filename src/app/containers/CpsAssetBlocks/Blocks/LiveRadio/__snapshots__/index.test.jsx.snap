// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPageBlocks LiveRadio should render correctly for amp 1`] = `
.c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

<html>
  <head>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    <div>
      <div
        class="c0"
      >
        <amp-iframe
          allowfullscreen="true"
          frameborder="0"
          layout="fill"
          sandbox="allow-scripts allow-same-origin"
          src="/ws/av-embeds/media/externalId/id"
        >
          <amp-img
            layout="fill"
            placeholder="true"
          />
        </amp-iframe>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPageBlocks LiveRadio should render correctly for canonical 1`] = `
.c1 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

.c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

<div
  class="c0"
>
  <iframe
    allow="autoplay; fullscreen"
    allowfullscreen=""
    class="c1"
    scrolling="no"
    src="/ws/av-embeds/media/externalId/id"
  />
</div>
`;
